### ISSUE TITLE
About how to use NDDS make own datasets

### ISSUE THREAD

**Issue by yehengchen** [Abdul-Mukit/dope_with_hand_tracking#1]
Hi,
I am trying to train DOPE with an own object.
I am also follow the NDDS Documentation and compelete the UE4 TestCapturer setting.
Could you tell me how can i make a new 3D object model and import it to UE4.
Thanks.
![Screenshot from 2019-08-29 21-49-22](https://user-images.githubusercontent.com/18131886/63943076-e8786980-caa9-11e9-8129-8541097c8da7.png)

**Comment by Abdul-Mukit** [comment]
Hi @yehengchen. You can make a 3D model in Blender then export it as a fbx file. Then import that fbx file into UE4. That should work. Sorry, for the later reply.

**Comment by Abdul-Mukit** [comment]
@yehengchen I don't know if these will help but you can check out my works with blender and ue4 in my youtube channel. I usually keep all my works/results documented there.
https://www.youtube.com/playlist?list=PLKyzdVp7WGd3P66eSMeAMD6kY5vkogY_S

**Comment by yehengchen** [comment]
@Abdul-Mukit Thanks for your reply.

I saw your youtube channel and make a 3D model using the Blender.
And I made screw dataset with randomize background (VOC images) using NDDS with the following figures.

<img width="366" alt="screw_cs" src="https://user-images.githubusercontent.com/18131886/64469419-77e9e080-d16c-11e9-8b50-2e0b3e5b23cf.png">
<img width="365" alt="screw_d" src="https://user-images.githubusercontent.com/18131886/64469420-79b3a400-d16c-11e9-99bb-a1a9f531acd2.png">
<img width="362" alt="screw_rgb" src="https://user-images.githubusercontent.com/18131886/64469422-7f10ee80-d16c-11e9-8c31-b1ec8015e1db.png">
<img width="365" alt="screw_json" src="https://user-images.githubusercontent.com/18131886/64469421-7cae9480-d16c-11e9-9e8d-71c2f9afc369.png">

Now I am facing the no detection problem.
I trained the model with 10000 simple images generated with NDDS with the following hyperparameters.

* Epochs: 100
* Batch Size: 16
* Learning Rate: 0.001

After training I used the .pth weights file and added new obj info to the DOPE config .yaml file then run the roslauch for DOPE, but I didn't get any output of screw pose in rviz.
Do you have any suggestions with how to train DOPE for small object (screw).
Thanks.

**Comment by Abdul-Mukit** [comment]
Did you try looking at the belief maps? In the original DOPE issues, I once gave the codes for visualizing belief maps. You can try that out to see if your network actually learned anything at all.

**Comment by yehengchen** [comment]
@Abdul-Mukit Do you think my "cuboid_dimensions": [ 86.780097961425781, 62.338001251220703, 416.13360595703125 ]"  is wrong?

**Comment by Abdul-Mukit** [comment]
Yes and no. If your 3D model is of that size then it's correct. But, even if they are, these are in cm units. Which means your 3D model is not realistic.

On another note, I think it should be very very difficult to track a tiny object as a screw. I am trying to detect a pen and it's already super hard. I highly doubt DOPE will be able to detect the pose of it.

If your application uses a camera that focuses on some product to see if the screws are properly aligned then the only way I can image DOPE being able to work is if the camera does a close-up shot with the screw on a plane/solid background.

**Comment by yehengchen** [comment]
Exactly! That is very hard to track a small targets. 
When I changed the pose of screw DOPE doesn't work.
![Screenshot from 2019-09-08 14-18-10](https://user-images.githubusercontent.com/18131886/64579096-934b2a80-d3bc-11e9-8c5d-cade1a8fc582.png)

**Comment by avinashsen707** [comment]
@yehengchen 
@Abdul-Mukit 

Sorry in advance for the interruption , if i get involved in the current issues!

I am beginer in DOPE ; started with traing already available FAT dataset of nvidia.
I ended up getting 10 weights while giving 10 epochs!
and i also got "heade.txt"
"Loss_train.csv"- which have a excel sheet of values!

What this all actually mean?? could you both help me!
i am adding a screenshot below..

Also i am looking which weight should be placed finally in the weights folder of DOPE?

Thanking You!
![Screenshot from 2020-02-04 13-02-00](https://user-images.githubusercontent.com/55526162/73634021-30697100-4686-11ea-9a15-59524d9bc2ad.png)
[header.txt](https://github.com/Abdul-Mukit/dope_with_hand_tracking/files/4146714/header.txt)

**Comment by Abdul-Mukit** [comment]
Sorry for the late reply. All your questions should be answered if you look at the train.py file. There is even a help option that should print out details of input arguments.
